### PR TITLE
JBOF: Just a Bunch of Fixes

### PR DESCRIPTION
**Pull request by RubenRBS**
- Renames orphancompiler.py script to propscheck.py, the name fits better
- Fixes Go props to make them easier to maintain
- Alphabetically sorts some listings in the code
- Adds tests to the props checker script (No idea how to integrate those so it's manual for now)
- Adds 2 missing limit cases for Motd filtering

### CODE REVIEW

File: etc/scripts/util/propschecktest.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import unittest
```
Comment (mattgodbolt) [inline etc/scripts/util/propschecktest.py:1]: Love it!

File: etc/scripts/util/test/cases/bad_compilers.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+compilers=a:b
+compiler.a.exe=/dev/null
```
Comment (mattgodbolt) [inline etc/scripts/util/test/cases/bad_compilers.properties:2]: are the "no newline at end" of all these files intentional? (github gets angry)

File: static/tests/motd.ts [hunk 3]
```diff
@@ -94,130 +94,156 @@ class MotdTests implements ITestable {
         );
 
         MotdTests.assertAdWithDateNow(
-            1641596400000, // 2022-01-08T00:00:00
+            Date.parse('2022-01-08T00:00:00Z'),
```
Comment (mattgodbolt) [inline static/tests/motd.ts:97]: oh my word SO MUCH BETTER thank you!

**Review by mattgodbolt — APPROVED** [review]
Thank you thank you!

File: etc/scripts/util/propscheck.py [hunk 4]
```diff
@@ -39,47 +41,85 @@
...
+
+    seen_lines = set()
+    duplicate_lines = set()
+
     disabled = set()
+
     with open(file) as f:
-        for line in f:
-            match_empty = EMPTY_LIST_RE.match(line)
+        line_number = 0
```
[the remark this answers is not in the document]
Comment (RubenRBS) [reply etc/scripts/util/propscheck.py:100]: Ooooh, fancy

File: etc/scripts/util/test/cases/bad_compilers.properties (hunk 2 above)
Comment (RubenRBS) [reply etc/scripts/util/test/cases/bad_compilers.properties]: Nope, fixed :)

File: etc/config/go.amazon.properties [hunk 5]
```diff
@@ -1,8 +1,7 @@
+compilers=&gccgo:&gl:&cross
```
[the remark this answers is not in the document]
Comment (RubenRBS) [reply etc/config/go.amazon.properties:1]: I'm not trusting myself, so I'll bring staging up to check that both prod's and staging's `/api/compilers/` return the same

[the remark this answers is not in the document]
Comment (RubenRBS) [reply etc/config/go.amazon.properties:1]: Checked, after fixing 1 issue it looks like before, so I guess lgtm

**Pull request merged**